packages/gamut-labs/src/landingPage/PageHero.tsx

```diff
@@ -14,49 +16,86 @@ const RightColumn = styled(Column)`
   }
 `;
 
+const VideoColumn = styled(Column)`
+  margin-left: 1rem;
+`;
+
 const Image = styled.img`
   width: 100%;
 `;
 
+const StyledVideo = styled(Video)`
+  align-self: center;
+`;
+
+export type ImageProps = {
+  src: string;
+  alt: string;
+};
+
+export type MediaProps =
+  | ({
+      type: 'image';
+    } & ImageProps)
+  | ({
+      type: 'video';
+    } & VideoProps);
+
 export type PageHeroProps = BaseProps & {
-  /**
-   * Hero image URL
-   */
-  imgSrc?: string;
-  /**
-   * Hero image alt text (for screen readers)
-   */
-  imgAlt?: string;
+  media?: MediaProps;
 };
 
 export const PageHero: React.FC<PageHeroProps> = ({
   title,
   desc,
   cta,
-  imgSrc,
-  imgAlt = '',
+  media,
   testId,
   onAnchorClick,
-}) => (
-  <LayoutGrid testId={testId}>
-    <Column
-      size={{
-        xs: 12,
-        sm: imgSrc ? 9 : 12,
-      }}
-    >
-      {title && <Title isPageHeading>{title}</Title>}
-      {desc && <Description text={desc} onAnchorClick={onAnchorClick} />}
-      {cta && (
-        <CTA href={cta.href} onCtaButtonClick={cta.onClick}>
-          {cta.text}
-        </CTA>
-      )}
-    </Column>
-    {imgSrc && (
-      <RightColumn size={3}>
-        <Image src={imgSrc} alt={imgAlt} />
-      </RightColumn>
-    )}
-  </LayoutGrid>
-);
+}) => {
+  const renderMedia = () => {
+    if (media && media.type === 'image') {
```

Comment:
TypeScript doesn't have any way of knowing that the `renderMedia` function would only be called when `media` exists, so it's forcing you to add these unnecessary `media &&` checks, right?

In general, having a `const render* = ...` method with hard-to-describe logic is a potential indicator that it might be a good idea to split the code out into a separate function. Maybe a `<MediaHero media={media} />` or something like that?

Alternately, I suppose you could move this function outside of the component and have it take in `media` as a parameter... if so:

```suggestion
    if (media?.type === 'image') {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
I'm going to opt for the first solution, I believe that will make the component more flexible when adding other types of media. I will move the `columnSize` function outside of the component